cmd/cloud-run/rotate-signify-certificate/rotate_signify_certificate.py

```diff
@@ -0,0 +1,279 @@
+        {
+            "csr": {
+                "value": csr.public_bytes(serialization.Encoding.PEM).decode("utf-8")
+            },
+            "validity": {"value": 7, "type": "DAYS"},
+            "policy": "sap-cloud-platform-clients",
+        }
+    )
+
+    cert_create_response = requests.post(
+        certificate_service_url,
+        headers={
+            "Authorization": f"Bearer {access_token}",
+            "Content-Type": "application/json",
+            "Accept": "applicaiton/json",
+        },
+        data=crt_create_payload,
+        timeout=10,
+    ).json()
+
+    pkcs7_certs = cert_create_response["certificateChain"]["value"].encode()
+
+    return pkcs7.load_pem_pkcs7_certificates(pkcs7_certs)
+
+
+def prepare_new_secret(
+    certificates: List[x509.Certificate],
+    private_key: rsa.RSAPrivateKey,
+    secret_data: Dict[str, Any],
+    created_at: str,
+) -> Dict[str, Any]:
+    """Prepare new secret data"""
+
+    # format certificates
+    certs_string = ""
+
+    for cert in certificates:
+        certs_string += f"subject={cert.subject.rfc4514_string()}\n"
+        certs_string += f"issuer={cert.issuer.rfc4514_string()}\n"
+        certs_string += f"{cert.public_bytes(Encoding.PEM).decode()}\n"
+
+    private_key_bytes = private_key.private_bytes(
+        Encoding.PEM, PrivateFormat.PKCS8, serialization.NoEncryption()
+    )
+
+    return {
+        "certData": base64.b64encode(certs_string.encode()).decode(),
+        "privateKeyData": base64.b64encode(private_key_bytes).decode(),
+        "createdAt": created_at,
+        "certServiceURL": secret_data["certServiceURL"],
+        "tokenURL": secret_data["tokenURL"],
+        "clientID": secret_data["clientID"],
+        "password": "",  # keep it empty to maintain structure, but indicate it's not needed
+    }
+
+
+def extract_message_data(pubsub_message: Any) -> Any:
+    """Extract secret rotate emssage from pubsub message"""
+    if pubsub_message["attributes"]["eventType"] != "SECRET_ROTATE":
+        # pylint: disable=broad-exception-raised
+        raise Exception("Unsupported event type")
+
+    data = base64.b64decode(pubsub_message["data"])
+    return json.loads(data)
+
+
+def decrypt_private_key(private_key_data: bytes, password: bytes) -> bytes:
+    """Decrypr encrypted private key"""
+    # pylint: disable=line-too-long
+    private_key = serialization.load_pem_private_key(private_key_data, password)
+
+    # pylint: disable=line-too-long
+    return private_key.private_bytes(
+        Encoding.PEM, PrivateFormat.PKCS8, serialization.NoEncryption()
+    )
+
+
+def fetch_access_token(
+    certificate: bytes, private_key: bytes, token_url: str, client_id: str
+) -> str:
+    """fetches access token from given token_url using certificate and private key"""
+    # Use temporary file for old cert and key because requests library needs file paths,
+    # it's not a security concern because the code is running in known environment controlled by us
```

Comment:
```suggestion
def fetch_access_token(
    certificate: bytes, private_key: bytes, secret_data: Dict[str, Any]
) -> str:
    """Fetches an access token using the old certificate and private key."""
    token_url = secret_data["tokenURL"]
    client_id = secret_data["clientID"]
```

Reply:
I disagree, we need that comment.
Also we shouldn';t relay on secret_data structure whenever it's possible.